test/src/lib/update-lock.spec.ts

```diff
@@ -108,40 +112,101 @@ describe('lib/update-lock', () => {
 		});
 	});
 
+	describe('cleanupLocks', () => {
+		it('should cleanup all leftover Supervisor-created lockfiles', async () => {
+			const BASE_PATH = '/mnt/root/tmp/balena-supervisor/services';
+
+			// Setup fake filesystem & stubs
+			mockFs({
+				[BASE_PATH]: {
+					'1234567': {
+						one: {
+							'updates.lock': mockFs.file({ uid: lockfile.NOBODY_UID }),
+						},
+						two: {
+							'updates.lock': mockFs.file({ uid: 0 }),
+						},
+					},
+					'7654321': {
+						three: {
+							'updates.lock': mockFs.file({ uid: lockfile.NOBODY_UID }),
+						},
+						four: {
+							'updates.lock': mockFs.file({ uid: 0 }),
+						},
+					},
+				},
+			});
+			stub(fsUtils, 'exec').resolves({
```

Comment:
This test doesn't seem terribly useful, since is basically checking that everything returned by this call (which breaks the black box of the implementation) is removed of the filesystem. The call could be anything which means we are not really testing the cleanup locks behavior. 

This might be an argument for using node operations to find instead of calling `find` as that way we will be able to test the state of the filesystem before and after as a black box.

Alternatively we can do 

```
find ${lockfileParentDir} -type f -user ${lockfile.NOBODY_UID} -name "*updates.lock" | xargs rm
```

on `entry.sh` and just forget about the testing maybe?

Reply:
Yeah, that's basically what I'm testing though - that anything returned by `find` is removed from the filesystem. Here we are working on the assumption that `find` always runs as expected, since this is a unit test.

I like the idea of running the `find` command in entry.sh instead though.

Reply:
Changed to `find "/mnt/root${BASE_LOCK_DIR}" -type f -user "${LOCKFILE_UID}" -name "*updates.lock" -delete` which avoids any errors with piping `find` to `rm`